src/hash/rpo/tests.rs

```diff
@@ -53,37 +52,31 @@ fn hash_elements_vs_merge() {
 }
 
 #[test]
-fn hash_elements_vs_merge_in_domain() {
+fn merge_vs_merge_in_domain() {
     let elements = [Felt::new(rand_value()); 8];
 
     let digests: [RpoDigest; 2] = [
         RpoDigest::new(elements[..4].try_into().unwrap()),
         RpoDigest::new(elements[4..].try_into().unwrap()),
     ];
 
-    // pick a random domain value.
-    let domain = Felt::new(rand_value());
-
-    // convert the elements into a list of base field elements
-    let elements = Felt::as_base_elements(&elements);
+    // ------------- merge with domain = 0 ----------------------------------------------------------
 
-    // initialize state to all zeros.
-    let mut state = [ZERO; STATE_WIDTH];
+    let merge_result = Rpo256::merge(&digests);
 
-    // set the second capacity element to the domain.
-    state[CAPACITY_RANGE.start + 1] = domain;
+    // set domain to zero. This should not change the result.
+    let domain = ZERO;
 
-    // absorb elements into the state.
-    state[RATE_RANGE.start..RATE_RANGE.end].copy_from_slice(elements);
+    let merge_in_domain_result = Rpo256::merge_in_domain(&digests, domain);
+    assert_eq!(merge_result, merge_in_domain_result);
 
-    // apply permutation to the state.
-    Rpo256::apply_permutation(&mut state);
+    // ------------- merge with domain != 0 ----------------------------------------------------------
 
-    // return the first 4 elements of the state as hash result
-    let h_result = RpoDigest::new(state[DIGEST_RANGE].try_into().unwrap());
+    // set domain to a random value. This should change the result.
+    let domain = Felt::new(rand_value());
```

Comment:
nit: I'd probably set `domain` to just `ONE` - not sure if setting it to a random value adds much benefit.